errors.go

```diff
@@ -49,6 +49,14 @@ func coupleAPIErrors(r *resty.Response, err error) (*resty.Response, error) {
 	}
 
 	if r.Error() != nil {
+		// Check that response is of the correct content-type before unmarshalling
+		expectedContentType := r.Request.Header.Get("Accept")
+		responseContentType := r.Header().Get("Content-Type")
+		if responseContentType != expectedContentType {
+			msg := fmt.Sprintf("Expected: %v; Received: %v", expectedContentType, responseContentType)
```

Comment:
It'd probably be better if this error message was something like `unexpected content type: expected %v, got %v` and if the test above checked for that.